src/main/scala/fi/oph/koski/valpas/opiskeluoikeusrepository/ValpasOpiskeluoikeusDatabaseService.scala

```diff
@@ -146,17 +147,36 @@ WITH
       ov_kelvollinen_opiskeluoikeus.henkilo_on_oppivelvollinen IS TRUE
       -- (1) oppijalla on peruskoulun opiskeluoikeus
       AND ov_kelvollinen_opiskeluoikeus.koulutusmuoto = 'perusopetus'
-      AND r_paatason_suoritus.suorituksen_tyyppi = 'perusopetuksenvuosiluokka'
-      -- (2) kyseisessä opiskeluoikeudessa on yhdeksännen luokan suoritus.
-      AND r_paatason_suoritus.koulutusmoduuli_koodiarvo = '9'
+      AND (
+        -- (2.1) kyseisessä opiskeluoikeudessa on yhdeksännen luokan suoritus.
+        (
+          r_paatason_suoritus.suorituksen_tyyppi = 'perusopetuksenvuosiluokka'
+          AND r_paatason_suoritus.koulutusmoduuli_koodiarvo = '9'
+        )
+        -- (2.2) tai oppija täyttää vähintään 17 vuotta tarkasteluvuonna: heidät näytetään luokka-asteesta riippumatta, koska voivat lopettaa
+        -- peruskoulun ja siirtyä seuraavaan opetukseen, vaikka olisivat esim. vasta 8. luokalla
+        OR (
+          ov_kelvollinen_opiskeluoikeus.henkilo_tayttaa_vahintaan_17_tarkasteluvuonna IS TRUE
+        )
+      )
       -- (3a) valvojalla on oppilaitostason oppilaitosoikeus ja opiskeluoikeuden lisätiedoista ei löydy kotiopetusjaksoa, joka osuu tälle hetkelle
       --      TODO (3b): puuttuu, koska ei vielä ole selvää, miten kotiopetusoppilaat halutaan käsitellä
       AND kotiopetusjaksoja.count = 0
-      -- (4)  opiskeluoikeus ei ole eronnut tilassa tällä hetkellä
       AND (
-        (aikajakson_keskella.tila IS NOT NULL AND NOT aikajakson_keskella.tila = any('{eronnut, katsotaaneronneeksi, peruutettu}'))
-        OR (aikajakson_keskella.tila IS NULL AND $tarkastelupäivä < ov_kelvollinen_opiskeluoikeus.alkamispaiva)
-        OR (aikajakson_keskella.tila IS NULL AND $tarkastelupäivä > ov_kelvollinen_opiskeluoikeus.paattymispaiva AND NOT ov_kelvollinen_opiskeluoikeus.viimeisin_tila = any('{eronnut, katsotaaneronneeksi, peruutettu}'))
+        -- (4,1) opiskeluoikeus ei ole eronnut tilassa tällä hetkellä
+        (
+          (aikajakson_keskella.tila IS NOT NULL AND NOT aikajakson_keskella.tila = any('{eronnut, katsotaaneronneeksi, peruutettu}'))
+          OR (aikajakson_keskella.tila IS NULL AND $tarkastelupäivä < ov_kelvollinen_opiskeluoikeus.alkamispaiva)
+          OR (aikajakson_keskella.tila IS NULL AND $tarkastelupäivä > ov_kelvollinen_opiskeluoikeus.paattymispaiva AND NOT ov_kelvollinen_opiskeluoikeus.viimeisin_tila = any('{eronnut, katsotaaneronneeksi, peruutettu}'))
```

Comment:
Näitä yli 120 merkkisiä rivejä voisi mahdollisuuksien mukaan pilkkoa. Hankala katsoa erityisesti Githubin pullarinäkymässä. :)

Reply:
siivoilen uutena committina viimisen pullarin perään.